### PR TITLE
Add support for Mongoid 7

### PR DESCRIPTION
**Pull request by kailan**
This pull request adds support for Mongoid 7 (whilst maintaining backwards compatibility) at the expense of removing [mongoid_paranoia](https://github.com/simi/mongoid_paranoia) due to its lack of support for the latest Mongoid version. Although this may be considered a blocker for releasing a new version, a quick look at the network tab for this repository shows many users forking the repository and performing hacky updates (breaking paranoia support along with other features) to allow their dependent applications to update to Mongoid 7, as paranoia is a feature used by very few users. For users that wish to continue using paranoia with older versions, they are still available. Existing applications using proper versioning in Gemfiles will not be affected.

This pull request also introduces the use of Mongoid::Compatibility `mongoid5_or_newer?` and `mongoid6_or_newer?` methods rather than strict version checks, allowing future non-breaking Mongoid releases to be supported without the need for updates.

Tests pass locally on all Mongoid versions from 3-7, although Travis is reporting failures for Mongoid 3 and 4 due to issues with the test environment. Running a Travis build on the current HEAD also fails for the same reasons, so this pull request is not the cause for these failures.

### CODE REVIEW

**Comment by dblock** [comment]
This looks good to me. Down with paranoia. We do need to fix the build, whether the failures are new in the test environment or not, before we get this merged.

**Comment by kailan** [comment]
Yeah no worries, I'm happy to look into that in the morning. Aiming for all
greens!

**Comment by kailan** [comment]
@dblock Tests passing on all versions 🎉

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
-## 5.3.4 (Next)
+## 6.0.0 (Next)
```
Comment (dblock) [inline CHANGELOG.md]: You also want to increment version.rb.

**Comment by dblock** [comment]
I'll take a closer look shortly. What was the issue?

**Comment by dblock** [comment]
Oh the ...`purge` thing again, got it...

**Comment by dblock** [comment]
I just merged this, great work.

Want to help co-maintain the gem @kailan? You could do the next release. Give me your rubygems email if so.

**Comment by kailan** [comment]
I would be love to @dblock 👍 

kailan@kmp.pw

**Comment by dblock** [comment]
Great, accept the invite at https://github.com/mongoid/mongoid-slug/invitations. Added you to Rubygems.

Opened https://github.com/mongoid/mongoid-slug/issues/253 for next release, thanks!

**Pull request merged**